### PR TITLE
Feature/item rendering properties

### PR DESCRIPTION
**Pull request by Nightenom**
Add posibility to override ItemProperties (things like clock/compass rotation), that otherwise needs full client level/are read from player entity.

### CODE REVIEW

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    protected static String getNextItemOverrides(final String str, final int startIndex)
+    {
+        return str.substring(startIndex, str.indexOf(')', startIndex));
+    }
+
+    /**
+     * Short call for adding itemProperty to current item
+     */
+    public void addPropertyForCurrentItem(final ResourceLocation propertyKey, final ClampedItemPropertyFunction property)
```
Comment (Thodor12) [inline src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java]: Any particular reason this uses `ClampedItemPropertyFunction`, and not just the interface `ItemPropertyFunction`?

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            .computeIfAbsent(Objects.requireNonNull(itemStack, "Call #setItem before this method").getItem(), item -> new HashMap<>())
+            .put(propertyKey, property);
+    }
+
+    /**
+     * @return modifiable all item-based overrides
+     */
+    public Map<Item, Map<ResourceLocation, ItemPropertyFunction>> getItemPropertyOverrides()
+    {
+        return itemPropertyOverrides;
```
Comment (Thodor12) [inline src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java:98]: Unmodifiable map maybe?

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    /**
+     * @return modifiable generic overrides
+     */
+    public Map<ResourceLocation, ItemPropertyFunction> getGenericPropertyOverrides()
+    {
+        return genericPropertyOverrides;
```
Comment (Thodor12) [inline src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java:106]: Unmodifiable map maybe?

**Review by Thodor12 — APPROVED** [review]

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java (hunk 3 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java:106]: intentionally modifiable, serves also as setter, basically the class doesnt care what's in the map

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java (hunk 1 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java]: no particular reason, if you want to add already created property you can use getMap().put, this method is merely intended as syntax sugar for quick addition of new/local instances and therefore requires the guarded (clamped) version instead

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java:106]: As we have an "add" function. Probably makes sense to be unmodifiable here

Comment (Nightenom) [reply src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java:106]: The add is only for non-generic and is pure syntax sugar (doesn't do anything special), I feel like unmodifiable makes little sense - you get full map methods vs we create all delegate method for both maps

**Comment by Nightenom** [comment]
Well, to address all three comments by Thodor:
both property maps are meant to be controlled by end-user and are completely fine and meant to be fully exposed via getters
the add method is purely syntax sugar for adding item specific to current itemStack
if we would return unmodifiable wrapper then we would have to add all map modifying methods to this class, and that doesnt make much sense to me
else suggests sth else

File: src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java (hunk 1 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockui/controls/ItemIconWithProperties.java]: fixed

**Pull request merged**